### PR TITLE
Allow trigger interval to be supplied as an argument

### PR DESCRIPTION
**Pull request by john-n-smith**
A signal counter is to be added to the sleeving line to count the number of punnets being sleeved.

The signal counter currently requires 300ms of signal to record a hit. This is too long for the sleeving line which operates at a rate > 200 punnets / minute.

The PR allows the trigger interval to be supplied as a command line argument.

### CODE REVIEW

File: src/signalCounter.c [hunk 1]
```diff
@@ -350,7 +350,7 @@ void signalIsr(void)
 
     printf("\n\nnew signal - interval was %llu\n", intervalTimeMs);
 
-    if( intervalTimeMs < TRIGGER_INTERVAL_MS) {
+    if( intervalTimeMs < triggerInterval) {
```
Comment (andyworsley) [inline src/signalCounter.c]: Unnecessary space

File: src/signalCounter.c [hunk 2]
```diff
@@ -372,13 +372,30 @@ void signalIsr(void)
...
     strcpy(endPointUrl, argv[1]);
-    fprintf(stderr, "Using [%s] as endpoint URL\n", endPointUrl);
+    printf("Using [%s] as endpoint URL\n", endPointUrl);
+
+    // store trigger interval, if we have one
+    if(argc == 3)
+    {
+        char* p; // will be set to the "first invalid character" set by strtol
+        errno = 0;
+        triggerInterval = strtol(argv[2], &p, 10);
```
Comment (andyworsley) [inline src/signalCounter.c:390]: EDITED: Is it an issue that `triggerInterval` it's defined as `short int`, but `strtol()` returns `long int`?

**Review by andyworsley — APPROVED** [review]

File: src/signalCounter.c (hunk 2 above)
Comment (john-n-smith) [reply src/signalCounter.c:390]: No, I've tested using huge ints and they get truncated. The actual (possibly truncated) value is reported the user on startup, so I don't think it's an issue.

Comment (john-n-smith) [reply src/signalCounter.c:390]: @andyworsley, I've switched the definition to `long int` for simplicity - it doesn't affect the operation of the program.

**Pull request merged**